docs: Document font manifest (#17463)
unoplatform/uno: doc/articles/features/custom-fonts.md

## Variable fonts and font manifest

Variable fonts are font files that can store multiple variants. They contain axes that define an aspect of design that can be varied. Common axes are width (`wdth`), weight (`wght`), italic (`ital`), and slant (`slnt`).

The width axis corresponds to `FontStretch` property in Uno Platform and WinAppSDK, the weight axis corresponds to the `FontWeight` property, and italic and slant axes correspond to the `FontStyle` property.

Variable fonts are currently properly supported on Android and Wasm, have partial support on iOS\*, and not supported on Skia due to [SkiaSharp issue](https://github.com/mono/SkiaSharp/issues/2318).

\* on Android and Wasm, if you set an aspect of a font that the variable font file doesn't have an axis for, the platform fakes it. For example, a variable font may have width and weight axes but not italic or slant. If you try to use italic, the platform will make "faux" italic. However, iOS doesn't have this behavior, which makes using variable fonts problematic in some cases.

> To overcome the issues for Skia and iOS, we introduced font manifest in Uno Platform 5.3. A font manifest is a json file that maps weight, stretch, and style to a static font. Note that font manifest is only supported on Skia and iOS because other platforms don't need it.

Reviewer: ```suggestion
To overcome the issues on Skia and iOS, we introduced the font manifest in Uno Platform 5.3. A font manifest is a JSON file that maps weight, stretch, and style to a static font. Note that the font manifest is only supported on Skia and iOS because other platforms don't need it.
```

### High level overview of how font manifest work

When you specify `FontFamily="ms-appx:///path/to/myfont.ttf`:

> - On iOS and Skia, we check if `ms-appx:///path/to/myfont.ttf.manifest` exists, then from there we will find the correct static font file to use.
> - On other platforms, `ms-appx:///path/to/myfont.ttf` will be used right away. When this file is a variable font, things will work as expected.

Reviewer: ```suggestion
- On iOS and Skia, we check if `ms-appx:///path/to/myfont.ttf.manifest` exists. From there, we find the correct static font file to use.
- On other platforms, `ms-appx:///path/to/myfont.ttf` will be used right away. When this file is a variable font, it will work as expected.
```
